mlflow/tracing/utils/__init__.py

```diff
@@ -447,3 +447,15 @@ def set_chat_attributes_special_case(span: LiveSpan, inputs: Any, outputs: Any):
 
     except Exception:
         pass
+
+
+def add_size_bytes_tag_to_trace(trace: Trace):
+    """
+    Calculate the size of the trace in bytes and add it as a tag to the trace.
+
+    This method modifies the trace object in place by adding a new tag.
+    """
+    prev_trace_size_bytes = len(trace.to_json().encode("utf-8"))
+    tag_size_bytes = len(str(prev_trace_size_bytes)) + len(MLFLOW_TRACE_SIZE_BYTES.encode("utf-8"))
+    trace_size_bytes = prev_trace_size_bytes + tag_size_bytes
```

Comment:
Is likely prone to at least an off-by-one error. Will see if there's a better approach.